zerver/actions/realm_settings.py

```diff
@@ -622,6 +639,7 @@ def do_delete_all_realm_attachments(realm: Realm, *, batch_size: int = 1000) ->
         obj_class._default_manager.filter(realm=realm).delete()
 
 
+@transaction.atomic
 def do_scrub_realm(realm: Realm, *, acting_user: UserProfile | None) -> None:
```

Comment:
This was suggested by @timabbott in [this comment](https://github.com/zulip/zulip/pull/26109#discussion_r1325231157).